### DISCUSSION TITLE
Tailwind Typography plugin styles overwritten by Preflight base styles

### DISCUSSION THREAD

**Discussion by rmill2016** [tailwindlabs/tailwindcss#10978] [Help] [answered]
Hello!

I came across an issue while working on  client project using Astro with Tailwindcss. I'm working with a CMS which is why I opted to use @tailwindcss/typography plugin for better styling. I noticed that the Preflight base styles font-size and font-weight are overwriting the manual styles I set for prose and styles under @layer base. Please refer to the minimal reproduction in Tailwind Play I have made. Thanks!

things I have tried:

- using prose-headings: selector to try and eliminate font-size and font-weight base styles
- setting all headers to font-size: unset; font-weight: unset; under @layer base directive
- Manually writting CSS without using @apply

Nothing has worked...

https://play.tailwindcss.com/ADXPqS6iKJ

**Comment by wongjn** [comment]
The `!important` marked styles are overriding the custom `h1` styling. Remove the `h1` selector from this rule and it should work as you'd like: https://play.tailwindcss.com/Ri0BegIGOx?file=css

**Reply by rmill2016** [reply]
Thank you! @wongjn those changes do fix it in tailwind play. Unfortunately applying those exact changes to development still doesn't work correctly. See screenshot
![hmmm](https://user-images.githubusercontent.com/97411281/230959124-a14ff332-927c-42ec-81a8-dd3fb00451b8.png)

**Reply by rmill2016** [reply]
Unchecking font-size: inherit and font-weight: inherit shows the correct styling. And for reference I put the
 h6,
  h5,
  h4,
  h3,
  h2,
  h1,
  p {
    font-size: unset;
    font-weight: unset;
  }
  
  styling at the very top inside @layer base {}

**Reply by wongjn** [reply]
It seems like you have two sources of CSS, one coming from an inline `<style>` element and other CSS coming from `globals.css` – this could be the cause of the discrepancy between your project and Tailwind Play.

**Reply by rmill2016** [reply]
Thanks for the quick reply! Here could be the inline styling in question that may affect it, though this is all the styling so I'm not sure if it could be the order I'm entering CSS in @layer base.

Index Page:
<figcaption class="absolute left-0 right-0 text-center layout top-1/2">
          <h2 class="uppercase">{homepage.headlines[0].content}</h2>
        </figcaption>
        
globals.css:
@layer base {
  html {
    @apply w-full h-[-webkit-fill-available] h-full overflow-x-hidden;
  }
  body {
    @apply w-full h-full max-w-full min-h-screen prose text-black prose-p:font-sans prose-h2:font-spartan prose-headings:m-0 prose-p:m-0 prose-ul:m-0 prose-img:m-0 prose-figure:m-0 prose-figcaption:m-0 prose-ol:m-0;
  }
  h4 {
    @apply font-sans text-2xl font-bold;
  }
  h3 {
    @apply font-sans text-3xl font-bold;
  }
  h2 {
    @apply text-4xl font-extrabold font-spartan;
  }
  h1 {
    @apply text-5xl font-extrabold font-spartan;
  }
  p {
    @apply font-sans text-lg font-normal;
  }
  ul {
    @apply list-disc;
  }
  li {
    @apply font-sans text-lg font-normal;
  }
  h6,
  h5,
  h4,
  h3,
  h2,
  h1,
  p {
    font-size: unset;
    font-weight: unset;
  }
}

That's everything so idk

**Reply by wongjn** [reply]
So you seem to be missing the `@tailwindcss` statements. These are needed to render Tailwind CSS rules. It also seems like you have two sources of CSS, one coming from an inline `<style>` element and other CSS coming from `globals.css` – this could be the cause of the discrepancy between your project and Tailwind Play.

**Reply by rmill2016** [reply]
Okay, sorry I didn't include the tailwind directives in my snippet, but they have been included. Also the inline styling is only uppercase, which should be applied on top of the base styles no?

**Reply by wongjn** [reply]
You misunderstand. If you look at your screenshot, you can see that you have two sources of CSS, one coming from an inline `<style>` element and other CSS coming from `globals.css` – this could be the cause of the discrepancy between your project and Tailwind Play.

![image](https://user-images.githubusercontent.com/11310624/230965292-93e155b6-0d4e-4da2-ad49-edb046d00256.png)

**Reply by rmill2016** [reply]
Thank you thank you thank you so much for your time in helping solve this frustrating problem. I took another screenshot, I completed commented out tailwind in order to minimize the CSS conflicting. As far as I understand tailwind shouldn't be applying any styles at this point so I'm really not sure where this other style is coming from. For context i'm using a fresh Astro project + Tailwind
![im-not-sure](https://user-images.githubusercontent.com/97411281/230967192-d18273d7-e123-4d18-8f4a-6b36e6b3c3c3.png)

**Reply by wongjn** [reply, marked as answer]
If you are running the project with a development server, it could be injected HMR CSS. You could try restarting the development server to see if that "clears" the outdated CSS. Otherwise, consider providing a remotely-hosted git repository of a project that reproduces the undesired behavior.

**Reply by rmill2016** [reply]
Hey update, I found the solution. So I completely forgot that in Astro how the CSS is loaded. I had to apply this config setting which was causing duplicate css issue. I will create a help thread in Astro. This was not a Tailwindcss issue, sorry!

astro.config.mjs file:

import { defineConfig } from 'astro/config'

import react from '@astrojs/react'
import tailwind from '@astrojs/tailwind'

// https://astro.build/config
export default defineConfig({
  integrations: [
    react(),
    tailwind({
      config: {
        applyBaseStyles: false,
      },
    }),
  ],
})
